aws/publisher.go

```diff
@@ -2,48 +2,61 @@ package aws
 
 import (
 	"context"
-	"log"
 
 	"github.com/aws/aws-sdk-go/aws"
 	"github.com/aws/aws-sdk-go/aws/session"
 	"github.com/aws/aws-sdk-go/service/sns"
 	"github.com/aws/aws-sdk-go/service/sns/snsiface"
 	pubsub "github.com/infobloxopen/atlas-pubsub"
+	"github.com/sirupsen/logrus"
 )
 
+type PublisherOption func(*publisher)
+
+func PublishWithLogger(logger *logrus.Logger) PublisherOption {
+	return func(pub *publisher) {
+		pub.logger = logger
+	}
+}
+
 // NewPublisher creates a new AWS message broker that will publish
 // messages to the given topic.
 // TODO: info on permissions needed within the config to make this work
 //
 // Topic names must be made up of only uppercase and lowercase
 // ASCII letters, numbers, underscores, and hyphens, and must be between 1 and
 // 247 characters long.
-func NewPublisher(sess *session.Session, topic string) (pubsub.Publisher, error) {
-	return newPublisher(sns.New(sess), topic)
+func NewPublisher(sess *session.Session, topic string, opts ...PublisherOption) (pubsub.Publisher, error) {
+	return newPublisher(sns.New(sess), topic, opts...)
 }
 
-func newPublisher(snsClient snsiface.SNSAPI, topic string) (*publisher, error) {
-	log.Printf("AWS: ensuring SNS topic exists for %q", topic)
+func newPublisher(snsClient snsiface.SNSAPI, topic string, opts ...PublisherOption) (*publisher, error) {
+	p := publisher{
+		sns:    snsClient,
+		logger: logrus.StandardLogger(),
```

Comment:
I see this logger will be overrided, but maybe it will be better to use toolkit logger with some default level (logrus.InfoLevel) as a default logger? It up to you, for me it seems more consistent

Reply:
I thought about this. I decided that using the `StandardLogger` seemed to make more sense as the default. This also allows using the global Logrus functions to change the behavior of this internal logging without having to make further changes (https://github.com/sirupsen/logrus/blob/master/exported.go#L14-L37)